addons/binding/org.openhab.binding.facerecognition/src/main/java/org/openhab/binding/facerecognition/handler/FaceRecognitionHandler.java

```diff
@@ -0,0 +1,310 @@
+    private final ChannelUID accessDeniedScoreChannelUID;
+    private final ChannelUID accessDeniedTimeChannelUID;
+    private final static String MODELS_FILE = "/var/lib/openhab2/etc/org.openhab.binding.facerecognition.models.xml";
+    private final static String FACE_CLASSIFIER_FILE = "/usr/share/opencv/haarcascades/haarcascade_frontalface_default.xml";
+    private ScheduledFuture<?> job;
+    private CascadeClassifier faceCascade;
+    private Mat accessDeniedFace;
+    private BigDecimal interval;
+    private BigDecimal threshold;
+
+    public FaceRecognitionHandler(Thing thing) {
+        super(thing);
+        httpClient = new HttpClient();
+        httpClient.setConnectTimeout(5000);
+
+        // Cache channel UIDs
+        imageChannelUID = new ChannelUID(thing.getUID(), CHANNEL_IMAGE);
+        accessGrantedImageChannelUID = new ChannelUID(thing.getUID(), CHANNEL_ACCESS_GRANTED_IMAGE);
+        accessGrantedUsernameChannelUID = new ChannelUID(thing.getUID(), CHANNEL_ACCESS_GRANTED_USERNAME);
+        accessGrantedScoreChannelUID = new ChannelUID(thing.getUID(), CHANNEL_ACCESS_GRANTED_SCORE);
+        accessGrantedTimeChannelUID = new ChannelUID(thing.getUID(), CHANNEL_ACCESS_GRANTED_TIME);
+        accessDeniedImageChannelUID = new ChannelUID(thing.getUID(), CHANNEL_ACCESS_DENIED_IMAGE);
+        accessDeniedUsernameChannelUID = new ChannelUID(thing.getUID(), CHANNEL_ACCESS_DENIED_USERNAME);
+        accessDeniedScoreChannelUID = new ChannelUID(thing.getUID(), CHANNEL_ACCESS_DENIED_SCORE);
+        accessDeniedTimeChannelUID = new ChannelUID(thing.getUID(), CHANNEL_ACCESS_DENIED_TIME);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Received channel: {}, command: {}", channelUID, command);
+
+        if (channelUID.equals(accessDeniedUsernameChannelUID)) {
+
+            // Train unknown face
+            if (command.getClass().equals(StringType.class) && !command.toString().isEmpty()
+                    && !command.toString().equalsIgnoreCase(UNKNOWN_USER_NAME) && (accessDeniedFace != null)) {
+
+                MatVector images = new MatVector(1);
+                images.put(0, accessDeniedFace);
+
+                // Setup label
+                Mat labels = new Mat(1, 1, CV_32SC1);
+                IntBuffer labelsBuffer = labels.createBuffer();
+                int label = command.toString().toLowerCase().hashCode();
+                labelsBuffer.put(0, label);
+
+                // Train face
+                faceRecognizer.update(images, labels);
+                faceRecognizer.setLabelInfo(label, command.toString());
+                faceRecognizer.save(MODELS_FILE);
+                logger.debug("Trained face of user \"{}\"", command.toString());
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing {}::{}", getThing().getLabel(), getThing().getUID());
```

Comment:
You don't need to log infront of an `updateStatus`. The framework will tell you about those events already